src/views/Settings.vue

```diff
@@ -118,13 +135,22 @@ export default {
         this.localApiEndpoint = value;
       }
     },
+    darkMode: {
+      get() {
+        return this.$store.state.settings.darkMode;
+      },
+      set(value) {
+        this.localDarkMode = value;
```

Comment:
You can avoid watching and dispatch directly in the setter

```suggestion
        this.$store.dispatch("updateTheme", value);
```